주소 필드 API 명세 맞춤 수정 및 다음 우편번호 API 연동

1000hyehyang opened the pull request:
## ⭐️ Issue Number

- #75

## 🚩 Summary

- 주소 필드명 API 명세에 맞게 수정: zipcode → zonecode, addressRoad → address1, addressJibun → address2로 변경하여 백엔드 API와 일치하도록 수정
- 주소 구조 변경: "도로명+지번" 구조에서 "기본주소+상세주소" 구조로 변경하여 API 문서에 맞게 수정
- 필드 길이 제한 업데이트: API 문서에 따른 필드 길이 제한 적용 (브랜드명 50자, 사업자명 100자, 대표이름 50자, 전화번호 20자, 주소 100자, 우편번호 5자리)
- 주문 페이지 레이아웃 개선: 3컬럼 그리드에서 단일 컬럼 세로 레이아웃으로 변경하여 사용자 경험 개선
- 주문 페이지 섹션 순서 최적화: 배송지 → 주문상품 → 포인트 → 결제요약 → 결제하기 버튼 순서로 변경
- 우편번호 입력 방식 개선: 표시 전용에서 입력 필드로 변경하여 5자리 숫자 제한 적용
- Daum 우편번호 API 연동: usePostcode 커스텀 훅 생성 및 판매자 회원가입, 배송지 관리, 주문 페이지에 우편번호 검색 기능 통합
- 자동 주소 입력 기능: 우편번호 검색 결과로 우편번호와 기본주소 자동 입력 기능 구현

## 🛠️ Technical Concerns

- 타입 안전성: 모든 주소 관련 타입 정의를 일관되게 수정하여 타입 안전성 확보
- 하위 호환성: 기존 주소 데이터 구조 변경으로 인한 마이그레이션 고려 필요
- 유효성 검증: 우편번호 5자리, 주소 필드 100자 제한 등 새로운 유효성 검증 규칙 적용
- 컴포넌트 재사용성: AddressFormFields, AddressCard 등 주소 관련 컴포넌트들의 일관된 필드명 사용
- 외부 API 의존성: Daum 우편번호 API 스크립트 자동 로드 및 에러 처리 고려 필요
- 팝업 보안: 우편번호 검색 팝업의 보안 설정 및 크로스 브라우징 이슈 고려


## 🙂 To Reviewer
@songcw8 
- [ ] 주소 필드명 변경이 API 명세와 일치하는지 확인 부탁드립니다
- [ ] 타입 정의 변경으로 인한 다른 컴포넌트들의 영향도 검토 부탁드립니다
- [ ] Daum 우편번호 API 연동이 안정적으로 동작하는지 테스트 부탁드립니다

## 📋 To Do


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **신규 기능**
  * 우편번호 검색 기능이 추가되어 주소 입력 시 Daum 우편번호 서비스를 사용할 수 있습니다.

* **버그 수정**
  * 주소 관련 입력 필드가 기존 `zipcode`, `addressRoad`, `addressJibun`에서 `zonecode`, `address1`, `address2`로 통일되어 입력 및 표시 오류가 개선되었습니다.
  * 판매자 회원가입 폼에서 상세주소(`address2`) 필수 입력 검증이 추가되었습니다.

* **UI/스타일**
  * 주문 페이지의 레이아웃이 간결해지고 결제 버튼이 새로 추가되었습니다.
  * 주문 페이지, 판매자 회원가입, 마이페이지 주소 카드 등에서 주소 입력 및 표시 레이아웃이 개선되고, 각 입력란의 최대 글자 수가 확장되었습니다.
  * 비밀번호 조건 표시에서 특수문자 목록이 명확해졌습니다.

* **기능 개선**
  * 사용자 보유 포인트가 5000에서 12345로 상향 조정되었습니다.
  * 비밀번호 특수문자 검증 패턴이 일부 특수문자로 제한되어 보안 정책이 강화되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

1000hyehyang committed:
fix: 주소 필드를 API 명세에 맞게 수정

- 주소 필드명 변경: zipcode → zonecode, addressRoad → address1, addressJibun → address2
- API 문서에 따른 필드 길이 제한 업데이트
- 주소 구조를 "도로명+지번"에서 "기본주소+상세주소"로 변경
- 우편번호 5자리 제한 추가
- 주소 필드 100자 제한 추가

1000hyehyang committed:
fix: 이전 PR에 변경되었던 UI 복구

1000hyehyang committed:
feat: 다음 우편번호 API 연동 기능 추가

- usePostcode 커스텀 훅 생성 및 자동 스크립트 로드
- 판매자 회원가입, 배송지 관리, 주문 페이지에 우편번호 검색 기능 통합
- 검색 결과로 우편번호와 기본주소 자동 입력
- 팝업 기반 직관적인 주소 검색으로 사용자 경험 개선

1000hyehyang committed:
fix: usePostcode 훅 개선 - 중복 스크립트 로딩 방지 및 의존성 경고 해결

- 중복 스크립트 로딩 방지를 위한 existingScript 체크 추가
- 스크립트 로딩 실패 시 에러 처리 추가
- React Hook 의존성 경고 해결을 위한 함수 순서 변경

1000hyehyang committed:
fix: 주소 필드명 일관되게 수정

1000hyehyang commented on src/app/order/page.tsx:
```diff
@@ -19,18 +19,35 @@ function OrderPageContent() {
-    zipcode: '',
-    addressRoad: '',
-    addressJibun: '',
+    zonecode: '',
+    address1: '',
+    address2: '',
```
수정했어

songcw8 reviewed (commented):
API 명세와 일치하며, 구조 및 길이 제한, 입력 방식, 자동입력, 우편번호 API 연동 등도 모두 명세에 맞게 구현되어 있습니다.
기존 main 코드와 관련하여 타입 정의 변경으로 인한 추가적인 영향이나 오류는 전혀 발생하지 않았습니다. 모든 코드가 새로운 주소 타입 명세(zonecode, address1, address2)를 일관되게 사용하고 있으며, 백엔드 API와도 호환됩니다.
테스트 해본 결과 다음 주소 연동 API도 동작하는 것 확인 했고 DB에도 반영되는 것 확인했습니다!
근데 한가지 제안사항은 제가 상세주소를 입력하지 않았을 때 회원가입이 안되는데 최상단 에러 메세지에는 "회원가입이 실패했습니다"만 나와서 사용자들이 무엇 때문에 실패를 한 부분인지 인식하는데 있어서 혼동이 있을 것 같아 이 부분도 "상세 주소가 누락되었습니다" 이렇게 메시지를 전달해주면 좋을 것 같습니다!

1000hyehyang committed:
feat: 판매자 회원가입 검증 및 일관성 개선

- 상세주소 검증 추가
- 주소 필드 명명 일관성 개선 (addressDetail → address2)
- 백엔드 비밀번호 패턴과 일치하도록 수정

1000hyehyang commented:
> API 명세와 일치하며, 구조 및 길이 제한, 입력 방식, 자동입력, 우편번호 API 연동 등도 모두 명세에 맞게 구현되어 있습니다. 기존 main 코드와 관련하여 타입 정의 변경으로 인한 추가적인 영향이나 오류는 전혀 발생하지 않았습니다. 모든 코드가 새로운 주소 타입 명세(zonecode, address1, address2)를 일관되게 사용하고 있으며, 백엔드 API와도 호환됩니다. 테스트 해본 결과 다음 주소 연동 API도 동작하는 것 확인 했고 DB에도 반영되는 것 확인했습니다! 근데 한가지 제안사항은 제가 상세주소를 입력하지 않았을 때 회원가입이 안되는데 최상단 에러 메세지에는 "회원가입이 실패했습니다"만 나와서 사용자들이 무엇 때문에 실패를 한 부분인지 인식하는데 있어서 혼동이 있을 것 같아 이 부분도 "상세 주소가 누락되었습니다" 이렇게 메시지를 전달해주면 좋을 것 같습니다!

감사합니다 창욱님!! 다시 검토해보니 백엔드에서 상세 주소가 필수라 그냥 입력이 안 되면 버튼을 못 누르게 해놨습니다..ㅎㅎ

1000hyehyang commented on src/constants/validation.ts:
```diff
@@ -4,7 +4,7 @@ export const VALIDATION_CONSTANTS = {
     MIN_LENGTH: 8,
     MAX_LENGTH: 50,
-    SPECIAL_CHARS: /[!@#$%^&*(),.?":{}|<>]/,
+    SPECIAL_CHARS: /[@$!%*#?&]/,
```
백엔드 검증과 일치시킨거야. 아직 실사용자가 없는 개발단계라 상관 없어

1000hyehyang committed:
feat: 주소 관련 검증 일관성을 위해 추가
